### ISSUE TITLE
object.Exception@src/itemdb.d(231): Unknow item id

### ISSUE THREAD

**Issue by obrobertsen** [skilion/onedrive#200]
Same as closed #141

**I have the same just compiled latest version today, here are the logfile:**
juni 15 18:27:01 mylaptop systemd[2061]: Started OneDrive Free Client.
juni 15 18:27:32 mylaptop onedrive[3406]: Moving remote item: ./file1.txt -> ./rename this file1.txt
juni 15 18:27:58 mylaptop onedrive[3406]: Uploading: ./hello.txt
juni 15 18:32:30 mylaptop onedrive[3406]: Deleting remote item: ./hello.txt
juni 15 18:33:05 mylaptop onedrive[3406]: object.Exception@src/itemdb.d(231): **Unknow item id**
juni 15 18:33:05 mylaptop onedrive[3406]: ----------------
juni 15 18:33:05 mylaptop onedrive[3406]: /usr/include/dmd/phobos/std/exception.d:388 pure @safe bool std.exception.enforce!(Except
juni 15 18:33:05 mylaptop onedrive[3406]: src/itemdb.d:231 immutable(char)[] itemdb.ItemDatabase.computePath(const(char)[], const(c
juni 15 18:33:05 mylaptop onedrive[3406]: src/sync.d:361 void sync.SyncEngine.deleteItems() [0x5db2cb]
juni 15 18:33:05 mylaptop onedrive[3406]: src/sync.d:185 void sync.SyncEngine.applyDifferences() [0x5d9ac7]
juni 15 18:33:05 mylaptop systemd[2061]: onedrive.service: Main process exited, code=killed, status=11/SEGV
juni 15 18:33:05 mylaptop systemd[2061]: onedrive.service: Unit entered failed state.
juni 15 18:33:05 mylaptop systemd[2061]: onedrive.service: Failed with result 'signal'.

**Comment by ibrokemypie** [comment]
```
2374F7E55D3E5741!138955 New.lrcat-journal
The item is marked for deletion
2374F7E55D3E5741!138954 New.lrcat.lock
The item is marked for deletion
2374F7E55D3E5741!111660 thermal-engine-msm8996.conf
The item is marked for deletion
2374F7E55D3E5741!111686 utils.c
object.Exception@src/itemdb.d(231): Unknow item id
----------------
??:? pure @safe bool std.exception.enforce!(Exception, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x58ee4d]
??:? immutable(char)[] itemdb.ItemDatabase.computePath(const(char)[], const(char)[]) [0x5c31c2]
??:? void sync.SyncEngine.applyDifference(std.json.JSONValue) [0x5d997d]
??:? void sync.SyncEngine.applyDifferences() [0x5d9404]
??:? void main.performSync(sync.SyncEngine) [0x5c4dab]
??:? _Dmain [0x5c460a]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5f1686]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5f15d0]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5f1642]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5f15d0]
??:? _d_run_main [0x5f1541]
??:? main [0x5c7d9d]
??:? __libc_start_main [0xd51b6510]
zsh: segmentation fault  onedrive -v

```
same here...

**Comment by mitchellmaler** [comment]
I am getting the same issue: object.Exception@src/itemdb.d(231): Unknow item id

**Comment by skilion** [comment]
Fixed in 849dd36276f99af87b36d7541bba79ba314d5b93

**Comment by ibrokemypie** [comment]
in latest commit still happening.
```
2017-07-06_12:48:55.16898 Loading config ...
2017-07-06_12:48:55.17013 Initializing the OneDrive API ...
2017-07-06_12:48:57.74707 Opening the item database ...
2017-07-06_12:48:57.74735 All operations will be performed in: /extra/home/onedrive
2017-07-06_12:48:57.74736 Initializing the Synchronization Engine ...
2017-07-06_12:48:57.74742 Applying differences ...
2017-07-06_12:49:00.09329 2374F7E55D3E5741!124 root
2017-07-06_12:49:00.09336 The item has not changed
2017-07-06_12:49:00.16336 2374F7E55D3E5741!29188 Pictures
2017-07-06_12:49:00.16345 The item has not changed
2017-07-06_12:49:00.25117 2374F7E55D3E5741!2033 Personal
2017-07-06_12:49:00.25126 The item has not changed
2017-07-06_12:49:00.35218 2374F7E55D3E5741!17751 Catalog
2017-07-06_12:49:00.35227 The item has not changed
2017-07-06_12:49:00.42999 2374F7E55D3E5741!57812 thingo
2017-07-06_12:49:00.43009 The item has not changed
2017-07-06_12:49:00.53274 2374F7E55D3E5741!42190 New
2017-07-06_12:49:00.53287 The item has not changed
2017-07-06_12:49:00.62770 2374F7E55D3E5741!136845 bumblebee-status
2017-07-06_12:49:00.62783 The item has not changed
2017-07-06_12:49:00.71994 2374F7E55D3E5741!136846 .git
2017-07-06_12:49:00.72008 The item has not changed
2017-07-06_12:49:00.79664 2374F7E55D3E5741!137136 New.lrcat.lock
2017-07-06_12:49:00.79677 The item is marked for deletion
2017-07-06_12:49:00.79678 2374F7E55D3E5741!137137 New.lrcat-journal
2017-07-06_12:49:00.79685 The item is marked for deletion
2017-07-06_12:49:00.79686 2374F7E55D3E5741!137127 index
2017-07-06_12:49:00.79694 The item is marked for deletion
2017-07-06_12:49:00.79694 2374F7E55D3E5741!137182 index.lock
2017-07-06_12:49:00.79702 The item is marked for deletion
2017-07-06_12:49:00.79703 2374F7E55D3E5741!137187 rebase-merge
2017-07-06_12:49:00.79710 The item is marked for deletion
2017-07-06_12:49:00.79711 2374F7E55D3E5741!137192 quiet
2017-07-06_12:49:00.79718 object.Exception@src/itemdb.d(231): Unknow item id
2017-07-06_12:49:00.79719 ----------------
2017-07-06_12:49:00.79819 ??:? pure @safe bool std.exception.enforce!(Exception, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x58e029]
2017-07-06_12:49:00.79821 ??:? immutable(char)[] itemdb.ItemDatabase.computePath(const(char)[], const(char)[]) [0x5c30df]
2017-07-06_12:49:00.79822 ??:? void sync.SyncEngine.applyDifference(std.json.JSONValue) [0x5d603d]
2017-07-06_12:49:00.79823 ??:? void sync.SyncEngine.applyDifferences() [0x5d5a6a]
2017-07-06_12:49:00.79824 ??:? void main.performSync(sync.SyncEngine) [0x5c4b8b]
2017-07-06_12:49:00.79825 ??:? _Dmain [0x5c4404]
2017-07-06_12:49:00.79826 ??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFNlZv [0x5eb976]
2017-07-06_12:49:00.79830 ??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5eb8c0]
2017-07-06_12:49:00.79833 ??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5eb932]
2017-07-06_12:49:00.79836 ??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5eb8c0]
2017-07-06_12:49:00.79836 ??:? _d_run_main [0x5eb83e]
2017-07-06_12:49:00.79837 ??:? main [0x5c82ad]
2017-07-06_12:49:00.79838 ??:? __libc_start_main [0xef930510]

```
